### PR TITLE
Separated normalization and presolve techniques

### PR DESCRIPTION
**Pull request by alexzucca90**
Fixes issue #78

### CODE REVIEW

File: dwave/preprocessing/include/dwave/presolve.h [hunk 1]
```diff
@@ -471,18 +473,18 @@ class Presolver {
 
 template <class bias_type, class index_type, class assignment_type>
 Presolver<bias_type, index_type, assignment_type>::Presolver()
-        : model_(), postsolver_(), default_techniques_(false), detached_(false) {}
+        : model_(), postsolver_(), default_techniques_(false), normalization_techniques_(false), detached_(false) {}
 
 template <class bias_type, class index_type, class assignment_type>
 Presolver<bias_type, index_type, assignment_type>::Presolver(model_type model)
-        : model_(std::move(model)), postsolver_(), default_techniques_(), detached_(false) {}
+        : model_(std::move(model)), postsolver_(), default_techniques_(), normalization_techniques_(), detached_(false) {}
```
Comment (arcondello) [inline dwave/preprocessing/include/dwave/presolve.h:480]:
```suggestion
        : model_(std::move(model)), postsolver_(), default_techniques_(false), normalization_techniques_(false), detached_(false) {}
```
might as well be consistent.

File: releasenotes/notes/separate-normalization-from-presolve-bb6116ffd3c25973.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Separated normalization techniques from other presolve techniques.
```
Comment (arcondello) [inline releasenotes/notes/separate-normalization-from-presolve-bb6116ffd3c25973.yaml:4]: Should mention the method name.

**Review by arcondello — COMMENTED** [review]
I think you exit out too early. Many of the other techniques, when applied once, are needed for normalization. Specifically
```
technique_remove_zero_biases
technique_remove_small_biases
technique_check_for_nan
```
all seem safe. We may even want to do
```
technique_remove_single_variable_constraints
```
and do something ad-hoc when they don't agree.

**Comment by alexzucca90** [comment]
Ok, agree on the techniques that I'm missing.
For `technique_remove_single_variable_constraints`, we could split the two parts of the function, the ones that tests for 0 variables constraints (that may raise an error) and the one for 1 variable constraints.

**Comment by arcondello** [comment]
Even the one variable constraints can have issues.  Consider
```
x <= 5
x >= 10
```

**Pull request closed without merging**